contract/contracts/NewInterfaces.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.6.0;
+
+contract DistributerInterface {
+    // TODO Add events
+
+    string name;
+    address tokenHolder;
+    mapping(address => address) tokenRefundDestinationList;
```

Comment:
I think that the token refunds will be carried out directly from Campaign to TokenHolder or Creator.
What is the role of the `tokenRefundDestinationList` in this Distributer?

Reply:
As we talked about yesterday, a campaign with an external token pattern is not used TokenHolder and should refund to external, so this state is necessary.

Reply:
Yes I know!
My question is why we need this variable in Distributer not in Campaign.
I think Campaign have tokens and transfer directly for Creator.

Reply:
Sorry I misunderstood.
First, tokens are transferred to Distributer by a creator directly or via TokenHolder using `Distributer#addToken()` function.
And then `Distributer#createCampaign()` is called and it is always called by a creator. 
Passing refund destination seems not good idea because a creator intentionally passes it as creator address instead of TokenHolder and when a campaign is cancelled, tokens could be refunded to a creator but not to TokenHolder.
So refund destination should determined by `msg.sender` with `addToken()` before `createToken()`, Distributer should manage refund address until campaign is created

Reply:
We use the following flow decided by meeting:
1. Approve token by creator or TokenHolder 
1. Send `createCampaign()` by a creator with address which is used as a transfer from
1. Manage address of token origin in campaign